pkg/network/setup/hotplugcontroller.go

```diff
@@ -20,9 +20,60 @@
 package network
 
 import (
+	"fmt"
+	"strings"
+
 	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
 )
 
+type ConcreteController struct {
+	ifaceCacheFactory cache.CacheCreator
+	vmi               *v1.VirtualMachineInstance
+	nsFactory         nsFactory
+}
+
+func NewInterfaceController(cacheFactory cache.CacheCreator, ns nsFactory) *ConcreteController {
```

Comment:
The correct thing, imo, is to have it idempotent: just invoke it again for all of them; for the ones already plugged in, there would be nothing to do. 

I think we may get this for free, since we return early on setup network phase#1 when the pod is already configured. Right ?

Reply:
Followed your suggestion.